### ISSUE TITLE
WebView malfunctions when "explorer.exe" is not running

### ISSUE THREAD

**Issue by verelpode** [MicrosoftEdge/WebViewFeedback#35]
Hello, is this "WebViewFeedback" repo strictly only for WebView2, or also for WebView1 bugs?  Should WebView1 bugs be posted to this repo or should I post to [https://github.com/Microsoft/microsoft-ui-xaml/](https://github.com/Microsoft/microsoft-ui-xaml/) or ...?

`Windows.UI.Xaml.Controls.WebView` has a wonderful option `WebViewExecutionMode.SeparateProcess` but it malfunctions when Windows Explorer (explorer.exe) is not running.  (I mean "explorer.exe" not "iexplore.exe".)

Firstly, create a WebView instance like this:
```
var myWebView = new Windows.UI.Xaml.Controls.WebView(WebViewExecutionMode.SeparateProcess);
testGridElement.Children.Add(myWebView);
...
myWebView.Navigate(new Uri("http://www.example.com"));
```

Either before or after creating the WebView instance at runtime, terminate explorer.exe.  For debugging purposes, you don't have to setup a real environment where explorer.exe does not run.  Instead you can simply perform tests by using this command to terminate explorer.exe:
```
taskkill /IM explorer.exe /F
```

Next, try to click the "X" to close/exit the UWP app that is using WebView.  Clicking the "X" does nothing.  The app refuses to close.  The associated "WWAHost.exe" process also remains running.  Thus, although UWP WebView does operate without explorer.exe, it malfunctions when you try to close/exit the app that is using WebView.

Re the order/sequence, I tested terminating explorer.exe before and after creating the WebView instance at runtime, and it seemed to make no difference.  Either way, the app failed to exit when clicking the "X" when explorer.exe was not running, regardless of when explorer.exe was terminated.

The bug is more severe when using the WPF version of WebView (`Microsoft.Toolkit.Wpf.UI.Controls.WebView`).  WPF WebView immediately **crashes** the whole app whenever explorer.exe is not running, whereas UWP WebView initially succeeds but then fails when you try to close/exit the app.  For the WPF version, I've already posted a report to: [https://github.com/windows-toolkit/Microsoft.Toolkit.Win32/issues/165](https://github.com/windows-toolkit/Microsoft.Toolkit.Win32/issues/165).

Reasons why explorer.exe is not necessarily running include:
* Remote Desktop Services, potentially, depending on configuration.
* Computers configured to operate in a kiosk mode.  Multiple kiosk methods exist, as follows.
* The ["Client-EmbeddedShellLauncher"](https://docs.microsoft.com/en-us/windows-hardware/customize/enterprise/shell-launcher) feature in Windows. 
* The registry key `Software\Microsoft\Windows NT\CurrentVersion\Winlogon` with `Shell` value configured, in order to launch a different .exe instead of explorer.exe. 
* The [kiosk / assigned access mode](https://docs.microsoft.com/en-us/windows/configuration/kiosk-single-app) feature that is configured via Start menu > Settings > Accounts.

Version:  I tested with a UWP app with the "Target Version" and "Min version" properties set to Windows 10 version 1903.

**Comment by david-risney** [comment]
This is just for the WebView2 control.

XAML or JavaScript WebView bugs you can provide feedback via the Windows 10 Feedback Hub app and that can be automatically hooked into the Windows bug database.

I believe you found the correct place for feedback on the Windows Community Toolkit's WebView control.

**Comment by liminzhu** [comment]
@verelpode Closing this and labeling as External as this is not a WebView2 issue. The channels @david-risney recommended would be the best avenues to surface the issue :).

**Comment by verelpode** [comment]
OK, thanks.  I hope WebView2 will be compatible with all of the aforementioned methods of creating a kiosk mode in Windows 10, and won't require explorer.exe to be running.  I also hope that when WebView2 is used in a separate-process mode, the web process will be launched in the same [desktop (HDESK)](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-createdesktopw) as the app that uses WebView2, in order to make WebView2 compatible with apps/processes running in desktops other than the default desktop (another kiosk mode issue).

**Comment by liminzhu** [comment]
Are you planning to use WebView2 from an UWP application (I'm not super familiar with Kiosk mode but I think only UWP apps work there)? Adding our partner @adambarlow whose team is building the XAML equivalent of WebView2 to surface the feedback.

**Comment by verelpode** [comment]
@liminzhu
> Are you planning to use WebView2 from an UWP application?

Yes, a UWP app, but also a WPF app.  Every part that can go in the UWP app, goes in the UWP app, and the rest in the WPF app.  Some parts cannot be placed in the UWP app because (1) not everything is supported in UWP, and (2) UWP sandbox restrictions.

> (I'm not super familiar with Kiosk mode but I think only UWP apps work there)?

Every type of Windows app works in one of the kiosk modes, but not every type of kiosk mode supports every app type.  Mainly the issue is that hopefully WebView2 will work regardless of whether "explorer.exe" is running or not.  This is not necessarily an ability that occurs automatically, as evidenced by the aforementioned problems that the current versions of `Microsoft.Toolkit.Wpf.UI.Controls.WebView` and `Windows.UI.Xaml.Controls.WebView` experience when "explorer.exe" is not running.

In other words, it's a problem in WebView 1, therefore it's worthwhile to ensure that WebView2 doesn't experience these problems, in both UWP and WPF apps. 

The related issue is the choice of desktop (HDESK) when WebView2 runs in a separate-process mode.   WebView 1 has a bug where it always launches the separate process in the default desktop, whereas it should launch the web process in the same desktop as the app that uses WebView.  Thus hopefully WebView2 will handle desktops correctly because it's relevant for kiosk modes.
